src/metamath.c

```diff
@@ -5150,6 +5150,86 @@ void command(int argc, char *argv[]) {
       print2("Bytes refer to compressed proof size, "
         "steps to uncompressed length.\n");
 
+      /* Clear tmp for all math tokens */
+      for (k = 0; k < g_mathTokens; k++)
+        g_MathToken[k].tmp = 0;
+
+      long step, mlen;
+      nmbrString *mString;
+
+      /* Set tmp to 1 for math tokens present in the proof */
+      m = nmbrLen(g_ProofInProgress.proof);
+      for (step = 0; step < m; step++) {
+        mString = (g_ProofInProgress.target)[step];
+        mlen = nmbrLen(mString);
+
+        for (i = 0; i < mlen; i++) {
+          g_MathToken[mString[i]].tmp = 1;
+        }
+      }
+
+      /* usefulStatements is a boolean array, where usefulStatements[k] == 1
+         iff statement k might be useful in minimization */
+      char *usefulStatements;
+      usefulStatements = poolMalloc(g_proveStatement*sizeof(char));
```

Comment:
Why not use `bool`, `true`, `false` as advised in https://github.com/metamath/metamath-exe/issues/17#issue-1091159614 ? Then a few statements below have minor modifications, e.g., `usefulCount += usefulStatements[k];` has to be replaced by something like `if (usefulStatements[k]) {usefulCount += 1}`

Reply:
I think originally I used `char` for consistency, for example, the `.tmp` field uses `0` and `1` instead of `bool`. As a positive side effect, `usefulCount` can be updated without branching (although I don't think it makes any great difference in this case).